client/src/progress-reporter.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import * as vscode from 'vscode';
+
+const EMPTY_DISPOSABLE = vscode.Disposable.from();
+
+class ProgressReporter implements vscode.Progress<unknown> {
+  private disposable: vscode.Disposable = EMPTY_DISPOSABLE;
```

Comment:
I was confused by what a "disposable" was until I looked at https://vscode-docs.readthedocs.io/en/stable/extensions/patterns-and-principles/#disposables. Could we name this `lastMessage` or `disposableMsg` or similar?

Reply:
yeah, disposables are prevalent in the vscode codebase. I guess it's the [RAII](https://en.wikipedia.org/wiki/Resource_acquisition_is_initialization) equivalent in Javascript.
I've renamed this `lastMessage`.

Reply:
I very much miss RAII in languages like JS and Python. But I digress..